internal/controller/pkg/revision/deployment.go

```diff
@@ -96,6 +96,9 @@ func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRe
 		s.Annotations = cc.Annotations
 		d.Labels = cc.Labels
 		d.Annotations = cc.Annotations
```

Comment:
I think we should not propagate metadata anymore now that they can give it via spec. Unrelated to this PR, we had a problem regarding claim->composite direct metadata propagation where ArgoCD used labels to identify claim but since composite had the same labels, it did its operations on that, too. So, I'd avoid direct propagation of metadata if possible due to its usage patterns in the ecosystem.

Reply:
Hmm this metadata is different than that we now include in the spec. These go to the `ServiceAccount` / `Deployment`, while the ones in the spec go to the `PodSpecTemplate`. We need to maintain the ability to add labels on the `ServiceAccount` because it is required for IRSA. I could be convinced to move these to the spec, but I currently like the model of keeping them separate because they act at kind of a "higher level". What do you think?

Reply:
Got it. It just feels like it's kind of surprising to get those propagated because when you add a metadata to an object, it belongs to that specific object and you use that metadata as [identifier](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) in some context or just for [documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) purposes for other tools. So, having them propagated down to different resources feels more like actual configuration, rather than metadata. From what I've seen, the operators that deploy multiple resources usually include specific configurations under spec, like `spec.serviceAccount`/`spec.serviceAccountTemplate` or `spec.deployment` etc. We could do a similar thing here as well if there is use case.

But as I pointed out, this isn't really relevant to this specific PR.

Reply:
@muvaf I tend to agree with you :+1: I'll see about addressing that in a subsequent PR